src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -0,0 +1,247 @@
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <fstream>
+
+#include "simulator.h"
+
+const float HRVO_TWO_PI        = 6.283185307179586f;
+const float ROBOT_RADIUS       = 0.09f;
+const float RADIUS_SCALE       = 1.0f;
+const int SIMULATOR_FRAME_RATE = 30;
+
+class HRVOTest : public ::testing::Test
+{
+   public:
+    Simulator simulator;
+
+    HRVOTest() : simulator()
+    {
+        simulator.setTimeStep(1.f / SIMULATOR_FRAME_RATE);
+        simulator.setAgentDefaults(/*neighborDist*/ 3.f, /*maxNeighbors*/ 30,
+                                   /*radius*/ ROBOT_RADIUS * RADIUS_SCALE,
+                                   /*goalRadius*/ 0.02f,
+                                   /*prefSpeed=*/3.5f, /*maxSpeed=*/4.825f,
+                                   /*uncertaintyOffset=*/0.f, /*maxAccel=*/3.28f);
+    }
+
+    void TearDown() override
+    {
+        run_simulator();
+    }
+
+    void create_div_b_field()
+    {
+        float field_width   = 9.f;
+        float field_height  = 6.f;
+        float robot_offsets = 2.1f * 0.25f;
+
+        for (float x = -(field_width / 2); x <= (field_width / 2); x += robot_offsets)
+        {
+            for (float y : {-(field_height / 2), field_height / 2})
+            {
+                const Vector2 position(x, y);
+                simulator.addAgent(position, simulator.addGoal(position), 1.f, 1, 0.25f,
+                                   0.25f, 0.1f, 0.1f, 0.f, 0.1f, Vector2(), 0.f);
+            }
+        }
+
+        // Vertical field lines
+        float max_y = (field_height / 2) - robot_offsets;
+        for (float y = -max_y; y <= max_y; y += robot_offsets)
+        {
+            for (float x : {-(field_width / 2), field_width / 2})
+            {
+                const Vector2 position(x, y);
+                simulator.addAgent(position, simulator.addGoal(position), 1.f, 1, 0.25f,
+                                   0.25f, 0.1f, 0.1f, 0.f, 0.1f, Vector2(), 0.f);
+            }
+        }
+    }
+
+    void add_static_obstacle(const Vector2 position, const float radius)
+    {
+        simulator.addAgent(position, simulator.addGoal(position), 1.f, 1, radius, radius,
+                           0.1f, 0.1f, 0.f, 0.1f, Vector2(), 0.f);
+    }
+
+    void run_simulator()
+    {
+        // The output file name is the name of the test
+        // The output is stored in the logging directory
+        std::string out_file_name(
+            ::testing::UnitTest::GetInstance()->current_test_info()->name());
+        std::ofstream output_file("/tmp/" + out_file_name + ".csv");
```

Comment:
we should be able to use TbotsGtestMain::logging_dir here

Reply:
Seems like `TbotsGtestMain::logging_dir` is defaulted to an empty string. Is that expected? I doubt we would want to store the outputs to root directory if no `logging_dir` argument is passed.

Reply:
yeah we could check for an empty string first I guess

Reply:
In the case of an empty string, should I use `/tmp` then? Or is there a different preferred directory?